### PR TITLE
Anomaly Core Production (but I actually finish it this time)

### PR DESCRIPTION
**Pull request by silicons**
Remake of #50528 

## About The Pull Request

https://hackmd.io/@tgstation/r1tzxpwPL

Basically that.
Raw anomaly cores
Cargo crates for them (currently 5k credits each, please tell me if this is good I balanced it relative to other crates I have no clue what the current cargo "meta"/power levels are)

- [x] Finish the explosive/implosion compressor attackby and implosion procs 

- [x] Map in the implosion compressor

- [x] Map in the roundstart 2 cores each type for science

Current values are set to maxcap at the 8th core, and 4 light radius (1/2/4 explosion required) at the first core. I think this is a bit low but I'm just going off the document.

**The implosion compressor consumes the TTV + tanks on an attempted detonation, regardless of success.** However, the "bare minimum" of checks are included (it won't let you insert incomplete TTVs. It will however, let you insert TTVs that have no explosive potential (test your bombs like you test your code ;) ), and will not waste the raw core if you fail to make a high enough radius.

Props to @ArcaneMusic for the machine sprite! 

## Why It's Good For The Game
image

Start of the anomaly core system I guess, gives toxins a use other than doing the quirky 50k maxcap tritium speedrun rush (or was that removed already?).

Yadda yadda allows locking items behind it and another method of getting these other than supermatter overloads (I don't know if that is already gone either by the by).

## Changelog
:cl:
add: An explosion compressor has been added to science. It makes raw anomaly cores into proper anomaly cores using prepared TTVs of a specific required blast radius.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Soon tm

**Comment by ArcaneMusic** [comment]
Cost-wise, 5k is still fairly expensive, but I'd say it's good for now and if it needs to be made cheaper that could always be changed down the line easily.

**Comment by MrDoomBringer** [comment]
Looks like there's some stray pixels in the anomaly cores

image

**Comment by kevinz000** [comment]
ahhhh fuck this is why i don't sprite :agony:
uhh gimme a sec

**Comment by kevinz000** [comment]
Yeah I can't see why this is happening on local.
Plus, I didn't even touch the old sprites other than renaming them to use _ instead of a blank space.
https://puu.sh/FTel7/6132f38603.gif

**Comment by kevinz000** [comment]
image
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

**Comment by kevinz000** [comment]
Conflict fixed, finally.

File: code/modules/research/anomaly/anomaly_core.dm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/obj/item/assembly/signaler/anomaly/manual_suicide(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user]'s [src] is reacting to the radio signal, warping [user.p_their()] body!</span>")
+	user.set_suicide(TRUE)
+	user.suicide_log()
+	user.gib()
+
+/obj/item/assembly/signaler/anomaly/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_ANALYZER)
+		to_chat(user, "<span class='notice'>Analyzing... [src]'s stabilized field is fluctuating along frequency [format_frequency(frequency)], code [code].</span>")
+	..()
```
Comment (Rohesie) [inline code/modules/research/anomaly/anomaly_core.dm:32]:
```suggestion
	return ..()
```

File: code/modules/research/anomaly/explosive_compressor.dm [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	. = ..()
+	if(istype(user) && user.Adjacent(src) && (user.mobility_flags & MOBILITY_USE))
+		if(!inserted_core)
+			to_chat(user, "<span class='warning'>There is no core inserted.</span>")
+			return
+		else
+			inserted_core.forceMove(get_turf(user))
+			to_chat(user, "<span class='notice'>You remove [inserted_core] from [src].</span>")
+			user.put_in_hands(inserted_core)
+			inserted_core = null
```
Comment (Rohesie) [inline code/modules/research/anomaly/explosive_compressor.dm:51]:
```suggestion
	if(!istype(user) || !user.Adjacent(src) || !(user.mobility_flags & MOBILITY_USE))
		return
	if(!inserted_core)
		to_chat(user, "<span class='warning'>There is no core inserted.</span>")
		return
	inserted_core.forceMove(get_turf(user))
	to_chat(user, "<span class='notice'>You remove [inserted_core] from [src].</span>")
	user.put_in_hands(inserted_core)
	inserted_core = null
```

File: code/modules/research/anomaly/explosive_compressor.dm [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	if(istype(I, /obj/item/raw_anomaly_core))
+		if(inserted_core)
+			to_chat(user, "<span class='warning'>There is already a core in [src].</span>")
+			return
+		if(!user.transferItemToLoc(I, src))
+			to_chat(user, "<span class='warning'>[I] is stuck to your hand.</span>")
+			return
+		inserted_core = I
+		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
+	else if(istype(I, /obj/item/transfer_valve))
```
Comment (Rohesie) [inline code/modules/research/anomaly/explosive_compressor.dm:94]:
```suggestion
		inserted_core = I
		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
		return
	if(istype(I, /obj/item/transfer_valve))
```

File: code/modules/research/anomaly/explosive_compressor.dm [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		// If they don't have a bomb core inserted, don't let them insert this. If they do, insert and do implosion.
+		if(!inserted_core)
+			to_chat(user, "<span class='warning'>There is no core inserted in [src]. What would be the point of detonating an implosion without a core?</span>")
+			return
+		var/obj/item/transfer_valve/V = I
+		if(!V.ready())
+			to_chat(user, "<span class='warning'>[V] is incomplete.</span>")
```
Comment (Rohesie) [inline code/modules/research/anomaly/explosive_compressor.dm:101]:
```suggestion
		var/obj/item/transfer_valve/valve = I
		if(!valve.ready())
			to_chat(user, "<span class='warning'>[valve] is incomplete.</span>")
```

**Comment by Rohesie** [comment]
Looking good, what's the status on this?
Tested locally and working?

**Comment by silicons** [comment]
uhhh

**Comment by optimumtact** [comment]
@silicons is this ready?

**Pull request merged**